File: server/src/main/java/ee/rmit/apps/dto/AppServiceResponse.java
package ee.rmit.apps.dto;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class AppServiceResponse {

    private final String serviceCode;
    private final String name;
    private final AppServiceType type;
    private final AppServiceSubType subType;
    private final String description;
    private final LocalDateTime lastModified;

    public AppServiceResponse(String serviceCode, String name, AppServiceType type, AppServiceSubType subType, String description, LocalDateTime lastModified) {
        this.serviceCode = serviceCode;
        this.name = name;
        this.type = type;
        this.subType = subType;
        this.description = description;
        this.lastModified = lastModified;
    }
}

File: client/src/server/dto/appserviceresponse.ts
import type { AppServiceType } from "@/server/dto/appservicetype";
import type { AppServiceSubType } from "@/server/dto/appservicesubtype";

export interface AppServiceResponse {
  serviceCode: string;
  name: string;
  type: AppServiceType;
  subType: AppServiceSubType;
  description: string;
  lastModified: string; // Assuming LocalDateTime is converted to a string
}